src/main/java/org/mockbukkit/mockbukkit/entity/ZombieVillagerMock.java

```diff
@@ -0,0 +1,80 @@
+package org.mockbukkit.mockbukkit.entity;
+
+import org.bukkit.OfflinePlayer;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Villager;
+import org.bukkit.entity.ZombieVillager;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.mockbukkit.mockbukkit.ServerMock;
+import org.mockbukkit.mockbukkit.exception.UnimplementedOperationException;
+
+import java.util.UUID;
+
+public class ZombieVillagerMock extends ZombieMock implements ZombieVillager
+{
+	private Villager.Type type = Villager.Type.PLAINS;
+	private Villager.Profession profession = Villager.Profession.NONE;
+	private OfflinePlayer conversionStarter = null;
+
+	/**
+	 * Constructs a new {@link ZombieMock} on the provided {@link ServerMock} with a specified {@link UUID}.
+	 *
+	 * @param server The server to create the entity on.
+	 * @param uuid   The UUID of the entity.
+	 */
+	public ZombieVillagerMock(@NotNull ServerMock server, @NotNull UUID uuid)
+	{
+		super(server, uuid);
+	}
+
+	@NotNull
+	@Override
+	public Villager.Type getVillagerType()
+	{
+		return type;
+	}
+
+	@Override
+	public void setVillagerType(@NotNull Villager.Type type)
+	{
+		this.type = type;
```

Comment:
I think the Preconditions.checkArgument was a better solution like suggested in the previous comment. Because this matches the paper behavior.
```java
Preconditions.checkArgument(type != null, "Villager.Type cannot be null");
```